subprojects/jacoco/src/main/java/org/gradle/internal/jacoco/JacocoAgentJar.java

```diff
@@ -93,6 +94,23 @@ public boolean apply(File file) {
         return !pre076;
     }
 
+    /** JVMs' above Java 8 need to have jdk.internal excluded, see https://github.com/gradle/gradle/issues/5184 */
+    public boolean includeNoLocationClassesNeedsJdkInternalExcluded() {
+        String jvmSpec = ManagementFactory.getRuntimeMXBean().getSpecVersion();
```

Comment:
If there is a better way to get the JVM spec version for the environment in which the agent will be run please call it out.